airgun/entities/cloud_insights.py

```diff
@@ -49,6 +60,93 @@ def update(self, values):
         view.fill(values)
 
 
+class RecommendationsTabEntity(BaseEntity):
+    endpoint_path = '/foreman_rh_cloud/insights_cloud'
+
+    def search(self, value):
+        """Search for 'query' and return matched hostnames/recommendations.
+
+        :param value: text to filter (default: no filter)
+        """
+        view = self.navigate_to(self, 'All')
+        time.sleep(5)
+        view.clear_button.click()
+        view.search_field.fill(value)
+        time.sleep(5)
+        return view.table.read()
+
+    def remediate_affected_system(self, recommendation_name, hostname):
+        """Open Affected systems, filter by hostname, select it, and click Remediate.
+
+        Returns the details view contents after remediation click.
+        """
+        # Use navigator to open the Affected Systems details view
+        view = self.navigate_to(self, 'Affected Systems', recommendation_name=recommendation_name)
+        view.search_field.wait_displayed()
+        # Filter by hostname and apply recommendation
+        view.search_field.fill(hostname)
+        wait_for(lambda: view.table.row(name=hostname), handle_exception=True, timeout=20)
+        time.sleep(15)
+        view.table[0][0].widget.click()
+        view.remediate.click()
+        self.browser.plugin.ensure_page_safe(timeout='30s')
+        modal = RemediateSummary(self.browser)
+        wait_for(lambda: modal.is_displayed, handle_exception=True, timeout=20)
+        modal.remediate.click()
+        view = JobInvocationStatusView(view.browser)
+        view.wait_for_result()
+        return view.read()
+
+    def bulk_remediate_affected_systems(self, recommendation_name):
+        """Open Affected systems, bulk select affected systems, and click Remediate.
+
+        Returns the details view contents after remediation click.
+        """
+        # Use navigator to open the Affected Systems details view
+        view = self.navigate_to(self, 'Affected Systems', recommendation_name=recommendation_name)
+        # view.search_field.wait_displayed()
+        wait_for(lambda: view.table.row(), handle_exception=True, timeout=20)
+        time.sleep(5)
```

Comment:
Can't we conceal these waits in the `Affected Systems` navigator `step` (line 138) function, make it more rigid there, so we don't have to deal with it in the entities where this navigator is used.